src/interpreter/ByteCode.cpp

```diff
@@ -16,21 +16,44 @@
 
 #include "Walrus.h"
 #include "interpreter/ByteCode.h"
-#include "interpreter/Opcode.h"
 #include "runtime/ObjectType.h"
 
 namespace Walrus {
 
-extern ByteCodeInfo g_byteCodeInfo[OpcodeKind::InvalidOpcode];
-
-static const uint8_t g_byteCodeSize[OpcodeKind::InvalidOpcode] = {
-#define WABT_OPCODE(rtype, type1, type2, type3, memSize, prefix, code, name, \
-                    text, decomp, size)                                      \
-    size,
-#include "interpreter/opcode.def"
-#undef WABT_OPCODE
+static const uint8_t g_byteCodeSize[ByteCode::OpcodeKindEnd] = {
+#define DECLARE_BYTECODE_SIZE(name) sizeof(name),
+    FOR_EACH_BYTECODE_OP(DECLARE_BYTECODE_SIZE)
+#undef DECLARE_BYTECODE_SIZE
+#define DECLARE_BYTECODE_SIZE(name, op, paramType, returnType) sizeof(name),
+        FOR_EACH_BYTECODE_BINARY_OP(DECLARE_BYTECODE_SIZE)
+            FOR_EACH_BYTECODE_UNARY_OP(DECLARE_BYTECODE_SIZE)
+#undef DECLARE_BYTECODE_SIZE
+#define DECLARE_BYTECODE_SIZE(name, op, paramType, returnType, T1, T2) sizeof(name),
+                FOR_EACH_BYTECODE_UNARY_OP_2(DECLARE_BYTECODE_SIZE)
+#undef DECLARE_BYTECODE_SIZE
 };
 
+ByteCode::ByteCode(ByteCode::Opcode opcode)
+#if defined(WALRUS_ENABLE_COMPUTED_GOTO)
+    : m_opcodeInAddress(g_byteCodeTable.m_addressTable[opcode])
+#else
+    : m_opcode(opcode)
+#endif
+{
+}
```

Comment:
Is this function is intentionally changed to no-inline?

Reply:
No, its just moved because `ByteCode::ByteCode` constructor needs `g_byteCodeTable` which is declared after `ByteCode` class. Shall I change to inline this constructor?

Reply:
Thanks for the quick review. 
I want to leave it as is now.